File: src/main/scala/study/alg/Palindrome.scala
package study.alg

object Palindrome extends App {

  def isPalindrome(x: Int): Boolean = {
    val s = x.toString
    s == s.reverse
  }

  def check(test: Array[Int]) {
    test.foreach(i => println(s"$i is palindrome: ${isPalindrome(i)}"))
  }

  check(Array(1, 2, 123, 121, 10000, 100001))
}

File: src/main/java/study/alg/strings/Palindrome.java
package study.alg.strings;

public class Palindrome {

    public static int isPalindrome(String A) {
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i< A.length(); i++) {
            char c = A.charAt(i);
            if (Character.isLetterOrDigit(c)) {
                sb.append(c);
            }
        }
        String s = sb.toString();
        //System.out.println(s);
        if (sb.reverse().toString().compareToIgnoreCase(s) == 0) {
            return 1;
        } else {
            return 0;
        }
    }

    public static void main(String[] args) {
        int p = isPalindrome("A man, a plan, a canal: Panama");
        System.out.println("is palindrome " + p);
    }
}
